ccvm_simulators/solvers/pumped_langevin_solver.py

```diff
@@ -0,0 +1,637 @@
+            ValueError: If the problem category is not supported by the solver.
+
+        Returns:
+            PumpedLangevinSolver: The PumpedLangevinSolver object.
+        """
+        super().__init__(device)
+        self.batch_size = batch_size
+        self.S = S
+        self._scaling_multiplier = LANGEVIN_SCALING_MULTIPLIER
+        # Use the method selector to choose the problem-specific methods to use
+        self._method_selector(problem_category)
+
+    @property
+    def parameter_key(self):
+        """The parameters that will be used by the solver when solving the problem.
+
+        Note:
+            Setting this parameter after calling tune() will overwrite tuned parameters.
+
+        The parameter_key must match the following format:
+
+            * key: problem size (the number of variables in the problem).
+            * value: dict with these keys:
+                * pump (float),
+                * dt (float),
+                * iterations (int),
+                * sigma (float),
+                * feedback_scale (float)
+
+            With values, the parameter key might look like this::
+
+                {
+                    20: {"dt": 0.005, "iterations": 15000, "sigma":0.02, "feedback_scale": 1.0}
+                }
+
+        Raises:
+            ValueError: If the parameter key does not contain the solver-specific
+                combination of keys described above.
+        """
+        return self._parameter_key
+
+    @parameter_key.setter
+    def parameter_key(self, parameters):
+        expected_dlparameter_key_set = set(
+            ["pump", "dt", "iterations", "sigma", "feedback_scale"]
+        )
+        parameter_key_list = parameters.values()
+        # Iterate over the parameters for each given problem size
+        for parameter_key in parameter_key_list:
+            if parameter_key.keys() != expected_dlparameter_key_set:
+                # The parameter key is not valid for this solver
+                raise ValueError(
+                    "The parameter key is not valid for this solver. Expected keys: "
+                    + str(expected_dlparameter_key_set)
+                    + " Given keys: "
+                    + str(parameter_key.keys())
+                )
+
+        # If we get here, the parameter key is valid
+        self._parameter_key = parameters
+        self._is_tuned = False
+
+    def _calculate_drift_boxqp(self, c, p, S):
+        """We treat the SDE that simulates the CIM of NTT as drift
+        calculation.
+
+        Args:
+            c (torch.Tensor): In-phase amplitudes of the solver
+            p (float): Instance of pump field
+            S (float): The saturation value of the amplitudes.
+
+        Returns:
+            tensor: The calculated change in the variable amplitude.
+        """
+
+        c_pow = torch.pow(c, 2)
+        c_drift_0 = torch.einsum("cj,cj -> cj", -1 + p - c_pow, c)
+
+        c_drift = c_drift_0 + self._calculate_grads_boxqp(c, S)
```

Comment:
@MehmetC-1QBit , here the `feedback_scale` should get multiplied by the gradient of the objective function and not the whole drift term (see my comment below). This wouldn't change your results as the simulation was done for `feedbackl_scale = 1`. But if its value is something else it will probably generate bad results. 
```suggestion
        c_drift = c_drift_0 + feedback_scale * self._calculate_grads_boxqp(c, S)
```
with `feedback_scale` having to be given to the `_calculate_drfit_boxqp` function, obviously.

Reply:
Done